### ISSUE TITLE
Use cached version of PHM-weight matrix after training.

### ISSUE THREAD

**Issue by tuanle618** [bayer-science-for-a-better-life/phc-gnn#11]
General idea: When training is finished, the PHM-weight matrix does not have to be constructed by a sum of Kronecker products in every forward pass. One could check for `self.train` and then use a cached version, e.g. saved in `self._cached_weight` of the final weight matrix for the linear transformation.
Using `torch.nn.functional.linear` function with inserting the `self._cached_weight` parameter.

Advantage: speedup gain in inference, as trainable parameters for PHM-layer are not updated, when running inference, we could just the final weight matrix, which is constructed by the optimized contribution and sub-weight matrices `{C_i}_{i=1}^{n}` and `{W_i}_{i=1}^{n}` , respectively.
